stackdriver/config/config_test.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"testing"
+)
+
+func TestNewStackdriverConfigFromConfigMap(t *testing.T) {
+	tests := []struct {
+		name           string
+		configMap      *corev1.ConfigMap
+		expectedConfig Config
+	}{
+		{
+			name: "fullSdConfig",
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"project-id":           "project",
+					"gcp-location":         "us-west1",
+					"cluster-name":         "cluster",
+					"gcp-secret-name":      "secret",
+					"gcp-secret-namespace": "non-default",
+				},
+			},
+			expectedConfig: Config{
+				ProjectID:          "project",
+				GcpLocation:        "us-west1",
+				ClusterName:        "cluster",
+				GcpSecretName:      "secret",
+				GcpSecretNamespace: "non-default",
+			},
+		},
+		{
+			name:           "emptySdConfig",
+			configMap:      &corev1.ConfigMap{},
+			expectedConfig: Config{},
+		},
+		{
+			name: "partialSdConfig",
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"project-id":   "project",
+					"gcp-location": "us-west1",
+					"cluster-name": "cluster",
+				},
+			},
+			expectedConfig: Config{
+				ProjectID:   "project",
+				GcpLocation: "us-west1",
+				ClusterName: "cluster",
+			},
+		},
+		{
+			name: "invalidGcpLocation",
```

Comment:
What does this test case mean?

Reply:
Hmm, it doesn't add much value in this test suite, I'll remove it.

There is the possibility that passing an invalid gcp location causes an error when creating Stackdriver exporter/client, but that codepath is not in this package.